docs/RANDOM_SEEDS.md

```diff
@@ -0,0 +1,416 @@
+   random bytes if we can, but neither delay boot nor trigger noisy kernel log
+   messages during early boot for these use-cases.
+
+## `systemd`'s Support for Filling the Kernel Entropy Pool
+
+systemd has various provisions to ensure the kernel entropy is filled during
+boot, in order to ensure the entropy pool is filled up quickly.
+
+1. When systemd's PID 1 detects it runs in a virtualized environment providing
+   the `virtio-rng` interface it will load the necessary kernel modules to make
+   use of it during earliest boot, if possible — much earlier than regular
+   kernel module loading done by `systemd-udevd.service`. This should ensure
+   that in VM environments the entropy pool is quickly filled, even before
+   systemd invokes the first service process — as long as the VM environment
+   provides virtualized RNG hardware (and VM environments really should!).
+
+2. The
+   [`systemd-random-seed.service`](https://www.freedesktop.org/software/systemd/man/systemd-random-seed.service.html)
+   system service will load a random seed from `/var/lib/systemd/random-seed`
+   into the kernel entropy pool. By default it does not credit entropy for it
+   though, since the seed is — more often than not — not reset when 'golden'
+   master images of an OS are created, and thus replicated into every
+   installation. If OS image builders carefully reset the random seed file
+   before generating the image it should be safe to credit entropy, which can
+   be enabled by setting the `$SYSTEMD_RANDOM_SEED` environment variable for
+   the service to `1`. Note however, that this service typically runs
+   relatively late during early boot: long after the initial RAM disk
+   (`initrd`) completed, and after the `/var/` file system became
+   writable. This is usually too late for many applications, it is hence not
+   advised to rely exclusively on this functionality to seed the kernel's
+   entropy pool. Also note that this service synchronously waits until the
+   kernel's entropy pool is initialized before completing start-up. It may thus
+   be used by other services as synchronization point to order against, if they
+   require an initialized entropy pool to operate correctly.
+
+3. The
+   [`systemd-boot`](https://www.freedesktop.org/software/systemd/man/systemd-boot.html)
+   EFI boot loader included in systemd is able to maintain and provide a random
+   seed stored in the EFI System Partition (ESP) to the booted OS, which allows
+   booting up with a fully initialized entropy pool from earliest boot
+   on. During installation of the boot loader (or when invoking [`bootctl
+   random-seed`](https://www.freedesktop.org/software/systemd/man/bootctl.html#random-seed))
+   a seed file when an initial seed is placed in `/loader/random-seed` in the
+   ESP. In addition, an identically sized randomized EFI variable is set (called
+   the 'system token'), which is written to the machine's firmware NVRAM. During
```

Comment:
"In addition, an identically sized randomized EFI variable called the 'system token' is written to the machine's firmware NVRAM."